### DISCUSSION TITLE
How to work with updating and using filters on custom fields?

### DISCUSSION THREAD

**Discussion by kharikbo** [prisma/prisma#18961] [Q&A] [answered]
Hi every one!

When updating, there is a need to use filtering not by unique fields, but by fields related with other tables. 

I would like the final request to be like this:

```
update table1 t set column1 = (select column1 from table2 where id = t.id)
or
update table1 t1 join table2 t2 on t1.id = t2.id set t1.column1 = t2.column1
```

Is it possible to do this? (without use $queryRaw)

**Comment by ludralph** [comment, marked as answer]
Hi @brights1de 👋 

Will something like this work for your use case. This snippet is updating a post through a user.
```js
let result = await prisma.user.update({
  where: { id: userId },
  data: {
    posts: {
      update: {
        where: { id: postId },
        data: { postDTO }
      }
    }
  }
})
```

**Reply by ludralph** [reply]
Hi @brights1de

Do you have any more questions about this issue? If not, it would be really helpful if you could quickly mark this **Discussion as answered** to indicate that your issue has been resolved.

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Reply by ludralph** [reply]
Hey @brights1de ,

do you have any more questions about this issue? If not, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!
